src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -4,13 +4,29 @@
 using System;
 using System.Diagnostics;
 using System.Numerics;
+using System.Runtime.CompilerServices;
 using System.Threading;
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Internal.Infrastructure
 {
     public struct MemoryPoolIterator
     {
+        // When this variable is set in KestrelEngine.JitReadonlyConsts it will 
+        // deterministically convert the following readonly static values to jitted consts 
+        // and embed the init code in that start up called function.
+        //
+        // This means when they are next used the have been pre-evaluated and the jit can 
+        // either directly embed them or use them for branch elimiation.
+        internal static bool StaticReadonlysJitted;
```

Comment:
The jit does the work when it first encounters the item. For readonly statics in a `beforefieldinit` class its when any static variable is first used [which is documented](https://msdn.microsoft.com/en-us/library/ms182275.aspx). 

That first function that is being jitted gets none of the benefits of the const like nature of the variables; though every function after that does. One of the features that doesn't happen is branch elimination - like on `IsLittleEndian`.

I don't know if its expected behaviour or a bug (might need to unwind and recomplile to resolve it). It came up before in https://github.com/aspnet/KestrelHttpServer/pull/554#r48912120